agent/src/main/java/com/teamscale/jacoco/agent/configuration/ConfigurationViaTeamscale.java

```diff
@@ -65,21 +65,21 @@ public static ConfigurationViaTeamscale retrieve(ILogger logger, String configur
 						processInformation).execute();
 			}
 			if (!response.isSuccessful()) {
-				if (response.code() >= 400 && response.code() < 500) {
-					throw new AgentOptionParseException(
-							"Failed to retrieve profiler configuration from Teamscale! " + response.errorBody()
-									.string());
-				}
 				throw new AgentOptionReceiveException(
-						"Failed to retrieve profiler configuration from Teamscale! " + response.errorBody().string());
+						"Failed to retrieve profiler configuration from Teamscale due to failed request. Http status: " + response.code()
+								+ " Body: " + response.errorBody().string());
 			}
 			ProfilerRegistration registration = response.body();
 			if (registration == null) {
-				throw new AgentOptionReceiveException("Failed to retrieve profiler configuration from Teamscale!");
+				throw new AgentOptionReceiveException(
+						"Failed to retrieve profiler configuration from Teamscale due to empty response. Status: " + response.code());
```

Comment:
I was wondering if we should include  `response.raw().body().string()` in the message. From the backend side I don't see how this could ever be `null`. So maybe Teamscale or a proxy etc. returns stuff here that just can't be deserialized correctly.

Reply:
I thought about that too, but by the time we catch the exception, the body will have been already consumed by the json parser, meaning this will give us a "closed" exception, right?

Reply:
AFAIU the body can only ever be read once :-( and we have no control over the json parsing doing so, since this happens automatically inside retrofit. So by the time we even get a response object, the body will already be gone

Reply:
good idea, implemented this :-) better be paranoid than sorry ;-)